x-pack/plugin/ccr/src/internalClusterTest/java/org/elasticsearch/xpack/ccr/RestartIndexFollowingIT.java

```diff
@@ -121,13 +116,31 @@ public void testFollowIndex() throws Exception {
     }
 
     private void setupRemoteCluster() throws Exception {
+        var remoteMaxPendingConnectionListeners = getRemoteMaxPendingConnectionListeners();
+
         ClusterUpdateSettingsRequest updateSettingsRequest = new ClusterUpdateSettingsRequest().masterNodeTimeout(TimeValue.MAX_VALUE);
         String address = getLeaderCluster().getAnyMasterNodeInstance(TransportService.class).boundAddress().publishAddress().toString();
         updateSettingsRequest.persistentSettings(Settings.builder().put("cluster.remote.leader_cluster.seeds", address));
         assertAcked(followerClient().admin().cluster().updateSettings(updateSettingsRequest).actionGet());
-        List<RemoteConnectionInfo> infos = followerClient().execute(RemoteInfoAction.INSTANCE, new RemoteInfoRequest()).get().getInfos();
-        assertThat(infos.size(), equalTo(1));
-        assertTrue(infos.get(0).isConnected());
+
+        if (remoteMaxPendingConnectionListeners == 1) {
+            // if queue is limited then update settings might fail to add a listener to wait for a connection
+            // however remote should be connected eventually by a concurrent task
+            assertBusy(this::isRemoteConnected);
+        } else {
+            assertTrue(isRemoteConnected());
+        }
+    }
+
+    private boolean isRemoteConnected() throws Exception {
+        var infos = followerClient().execute(RemoteInfoAction.INSTANCE, new RemoteInfoRequest()).get().getInfos();
+        return infos.size() == 1 && infos.get(0).isConnected();
+    }
+
+    private Integer getRemoteMaxPendingConnectionListeners() {
+        var response = followerClient().admin().cluster().prepareNodesInfo().setSettings(true).get();
```

Comment:
By default we request everything, so if you only want settings you have to clear them first:

```suggestion
        var response = followerClient().admin().cluster().prepareNodesInfo().clear().setSettings(true).get();
```

Also all the nodes have their own settings so you can further restrict to the coordinating node only:

```suggestion
        var response = followerClient().admin().cluster().prepareNodesInfo("_local").clear().setSettings(true).get();
```